### ISSUE TITLE
Helidon Rest Client giving "java.util.NoSuchElementException: No value present" when Retry is used

### ISSUE THREAD

**Issue by AchintoBanerjee** [helidon-io/helidon#5064]
## Environment Details
* Helidon Version: 2.5.1
* Helidon SE or Helidon MP : Helidon MP 
* JDK version: 17.0.3.1
* OS: macOS
* Docker version (if applicable): N/A

----------

## Problem Description
Helidon Rest Client is giving an exception "java.util.NoSuchElementException: No value present" when @Retry is used in the the Rest Client . This is easily reproduced on Helidon Quickstart prj . 

Stack Trace : 
```
java.util.NoSuchElementException: No value present
	at java.base/java.util.Optional.orElseThrow(Optional.java:377)
	at io.helidon.microprofile.faulttolerance.MethodIntrospector.<init>(MethodIntrospector.java:68)
	at io.helidon.microprofile.faulttolerance.CommandInterceptor.interceptCommand(CommandInterceptor.java:52)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.weld.interceptor.reader.SimpleInterceptorInvocation$SimpleMethodInvocation.invoke(SimpleInterceptorInvocation.java:73)
	at org.jboss.weld.interceptor.proxy.WeldInvocationContextImpl.invokeNext(WeldInvocationContextImpl.java:92)
	at org.jboss.weld.interceptor.proxy.WeldInvocationContextImpl.proceed(WeldInvocationContextImpl.java:124)
	at org.jboss.weld.bean.InterceptorImpl.intercept(InterceptorImpl.java:105)
	at org.glassfish.jersey.microprofile.restclient.InterceptorInvocationContext$InvocationInterceptor.intercept(InterceptorInvocationContext.java:136)
	at org.glassfish.jersey.microprofile.restclient.InterceptorInvocationContext.proceed(InterceptorInvocationContext.java:108)
	at org.glassfish.jersey.microprofile.restclient.RestClientModel.invokeMethod(RestClientModel.java:69)
	at org.glassfish.jersey.microprofile.restclient.ProxyInvocationHandler.invoke(ProxyInvocationHandler.java:70)
	at jdk.proxy2/jdk.proxy2.$Proxy93.getEntityResponseOnlyData(Unknown Source)
	at io.helidon.examples.quickstart.mp.rest.client.FaRestClientFacadeImpl.getEntityResponse(FaRestClientFacadeImpl.java:47)
	at io.helidon.examples.quickstart.mp.rest.client.FaRestClientFacadeImpl$Proxy$_$$_WeldClientProxy.getEntityResponse(Unknown Source)
	at io.helidon.examples.quickstart.mp.AnotherRestResource.createResponse(AnotherRestResource.java:40)
	at io.helidon.examples.quickstart.mp.AnotherRestResource.debugIssue(AnotherRestResource.java:36)
	at io.helidon.examples.quickstart.mp.AnotherRestResource$Proxy$_$$_WeldSubclass.debugIssue$$super(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.weld.interceptor.proxy.TerminalAroundInvokeInvocationContext.proceedInternal(TerminalAroundInvokeInvocationContext.java:51)
	at org.jboss.weld.interceptor.proxy.AroundInvokeInvocationContext.proceed(AroundInvokeInvocationContext.java:78)
	at io.helidon.servicecommon.restcdi.InterceptionRunnerImpl.run(InterceptionRunnerImpl.java:103)
	at io.helidon.servicecommon.restcdi.HelidonInterceptor$WithPostCompletion.aroundInvocation(HelidonInterceptor.java:180)
	at io.helidon.servicecommon.restcdi.HelidonInterceptor$Base.aroundInvoke(HelidonInterceptor.java:137)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.jboss.weld.interceptor.reader.SimpleInterceptorInvocation$SimpleMethodInvocation.invoke(SimpleInterceptorInvocation.java:73)
	at org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.executeAroundInvoke(InterceptorMethodHandler.java:84)
	at org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.executeInterception(InterceptorMethodHandler.java:72)
	at org.jboss.weld.interceptor.proxy.InterceptorMethodHandler.invoke(InterceptorMethodHandler.java:56)
	at org.jboss.weld.bean.proxy.CombinedInterceptorAndDecoratorStackMethodHandler.invoke(CombinedInterceptorAndDecoratorStackMethodHandler.java:79)
	at org.jboss.weld.bean.proxy.CombinedInterceptorAndDecoratorStackMethodHandler.invoke(CombinedInterceptorAndDecoratorStackMethodHandler.java:68)
	at io.helidon.examples.quickstart.mp.AnotherRestResource$Proxy$_$$_WeldSubclass.debugIssue(Unknown Source)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167)
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219)
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:475)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:397)
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:81)
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248)
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274)
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244)
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265)
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234)
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:684)
	at io.helidon.webserver.jersey.JerseySupport$JerseyHandler.lambda$doAccept$4(JerseySupport.java:327)
	at io.helidon.common.context.Contexts.runInContext(Contexts.java:117)
	at io.helidon.common.context.ContextAwareExecutorImpl.lambda$wrap$7(ContextAwareExecutorImpl.java:154)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
```

## Steps to reproduce
1. Run the attached Prj in debug mode preferably 
2. Invoke the curl -X GET http://localhost:8080/another
3. Notice the error as response 
4. This error is gone if I remove @retry from the Rest client and actual response from underlying rest is fetched. 

## Reproducible steps and code to look at 
Code : Refer attached Quickstart application where issue is reproduced 
Code : Walkthrough : Refer readme.md in the attached prj 

[rest_client_issue.zip](https://github.com/helidon-io/helidon/files/9700688/rest_client_issue.zip)

**Comment by spericas** [comment]
@AchintoBanerjee This one is actually fixed in 2.5.2. PR #4641. Could you try it out?

**Comment by AchintoBanerjee** [comment]
@spericas  - This seems to be working with 2.5.2 .
